src/univariate/continuous/normal.jl

```diff
@@ -1,12 +1,18 @@
-immutable Normal <: ContinuousUnivariateDistribution
-    μ::Float64
-    σ::Float64
+immutable Normal{T<:Number} <: ContinuousUnivariateDistribution
+    μ::T
+    σ::T
 
-    Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); new(μ, σ))
-    Normal(μ::Real) = Normal(μ, 1.0)
-    Normal() = Normal(0.0, 1.0)
+    Normal(μ::T, σ::T) = new(μ, σ)
 end
 
+Normal{T<:Number}(μ::T, σ::T) = Normal{T}(μ, σ)
+Normal(μ::Number, σ::Number) = Normal(promote(μ, σ)...)
+Normal(μ::Number) = Normal(μ, one(μ))
+#Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); Normal(μ, σ))
+Normal(μ::Real) = Normal(μ, one(μ))
+Normal() = Normal(0.0, 1.0)
+Normal(μ::Complex, σ::Complex) = error("Normal with complex mean and variance not defined")
```

Comment:
The algebraic insight that @StefanKarpinski gave makes me feel convinced that there is a reasonable way of viewing dual or forward diff numbers as some sort of real number. At the same time, the technical concerns that @goretkin raised make me wonder if we will end up producing erroneous parametric number types that are not so welcome.

I am not so sure myself what is the ideal solution here... I would only say, @jrevels, it may be better to wait a bit before making this drastic coding change to treat duals or forward diff numbers as a subtype of real till the discussion here matures and a consensus is reached.